FunctionalTableData/CollectionView/FunctionalCollectionData.swift

```diff
@@ -456,6 +458,19 @@ extension FunctionalCollectionData: UICollectionViewDataSource {
 		
 		sections[sourceIndexPath.section].didMoveRow?(sourceIndexPath.item, destinationIndexPath.item)
 	}
+	
+	public func collectionView(_ collectionView: UICollectionView, viewForSupplementaryElementOfKind kind: String, at indexPath: IndexPath) -> UICollectionReusableView {
+		// We expose this delegate method by requiring a method to get the reuseIdentifier, and then using that reuseIdentifier to
+		// dequeue a UICollectionReusableView, which can then have its appropriate attributes applied.
+		// This approach is required because `UICollectionViewDataSource` requires that this always return a valid view, and
+		// suggests skipping applying attributes to the view when a supplementary view is not wanted.
+		//
+		// The owner of the collectionView is still responsible for registering the supplementary element with `UICollectionView.register(_:forSupplementaryViewOfKind:withReuseIdentifier:)`
+		let reuseIdentifier = supplementaryElementReuseIdentifier?(collectionView, kind, indexPath) ?? ""
+		let view = collectionView.dequeueReusableSupplementaryView(ofKind: kind, withReuseIdentifier: reuseIdentifier, for: indexPath)
+		prepareViewForSupplementaryElement?(collectionView, view, kind, indexPath)
+		return view
```

Comment:
How do you intend to use this? I thought you wanted to have something like `TableHeaderFooterStateType` for UICollectionViews. This will only expose the guts of UICollectionDelegate to the view controller right? which goes "against" this library.

What are you going to be using the header for? (You can take this privately if its about implementation details)